### PR TITLE
Change draft colour based on allowed state

### LINKED ISSUE #142: Change color (e.g. red / green) of drafts based on possibility to build them on the spot
This is a follow up after #23.

Users need to immediately see whether a building can be build on a spot. Change color of the buildings to semi-transparent red/green based on collisions with other objects, out-of-map constraint, and so on.

### PR DESCRIPTION
**Pull request by 0HyperCube**
Closes #142

### CODE REVIEW

File: crates/spawner/src/draft.rs [hunk 1]
```diff
@@ -126,3 +142,69 @@ fn update_draft(
...
+#[derive(Clone, Resource)]
+struct DraftMaterials {
+    valid_placement: Handle<StandardMaterial>,
+    invalid_placement: Handle<StandardMaterial>,
+}
+
+fn insert_materials(mut commands: Commands, mut materials: ResMut<Assets<StandardMaterial>>) {
+    commands.insert_resource(DraftMaterials {
+        valid_placement: materials.add(Color::rgba(0.2, 0.8, 0.2, 0.7).into()),
+        invalid_placement: materials.add(Color::rgba(0.86, 0.08, 0.24, 0.7).into()),
```
Comment (0HyperCube) [inline crates/spawner/src/draft.rs:156]: Should these colours be declared as constants somewhere?

File: crates/spawner/src/draft.rs [hunk 2]
```diff
@@ -126,3 +142,69 @@ fn update_draft(
...
+    mut standard_materials: Query<&mut Handle<StandardMaterial>>,
+    scene_spawner: Res<SceneSpawner>,
+    draft_materials: Res<DraftMaterials>,
+) {
+    for (draft, children, mut draft_highlight) in &mut draft_query {
+        let allowed = draft.allowed();
+        // Skip assigning material if the allowed state hasn't changed
+        if draft_highlight.previously_allowed == Some(allowed) {
+            continue;
+        }
```
Comment (0HyperCube) [inline crates/spawner/src/draft.rs:191]: I'm struggling to get this working with change detection as for the first few frames after the draft is spawned, the scene doesn't exist.

File: crates/spawner/src/draft.rs (hunk 1 above)
Comment (Indy2222) [reply crates/spawner/src/draft.rs:156]: Yes, I would create constants at the top of the file.

File: crates/spawner/src/draft.rs [hunk 3]
```diff
@@ -126,3 +142,69 @@ fn update_draft(
...
+        invalid_placement: materials.add(Color::rgba(0.86, 0.08, 0.24, 0.7).into()),
+    });
+}
+
+// Assign the appropriate allowed to all entities in the spawned glb scene
+fn update_object_material(
+    allowed: bool,
+    entities: impl Iterator<Item = Entity>,
+    standard_materials: &mut Query<&mut Handle<StandardMaterial>>,
+    draft_materials: &Res<DraftMaterials>,
```
Comment (Indy2222) [inline crates/spawner/src/draft.rs:165]: `&Res<DraftMaterials>` -> `&DraftMaterials` would be nicer (you can get it with `res.as_ref()`.

File: crates/spawner/src/draft.rs (hunk 2 above)
Comment (Indy2222) [reply crates/spawner/src/draft.rs:191]: I see. I would create a ticket for scene pre-loading (e.g. during loading stage) and reference the ticket here in a comment. Using change detection would be nicer.

**Review by Indy2222 — CHANGES_REQUESTED** [review]

File: crates/spawner/src/draft.rs (hunk 2 above)
Comment (0HyperCube) [reply crates/spawner/src/draft.rs]: I managed to get the change detection working by adding a system that checks with a `SceneInstance` component is added.

File: crates/spawner/src/draft.rs [hunk 4]
```diff
@@ -126,3 +143,71 @@ fn update_draft(
...
+
+// Assign the appropriate allowed to all entities in the spawned glb scene
+fn update_object_material(
+    allowed: bool,
+    entities: impl Iterator<Item = Entity>,
+    standard_materials: &mut Query<&mut Handle<StandardMaterial>>,
+    draft_materials: &DraftMaterials,
+) {
+    for entity in entities {
+        let Ok(mut material_handle) = standard_materials.get_mut(entity) else{
```
Comment (Indy2222) [inline crates/spawner/src/draft.rs:169]: There is a forgotten space after `else`.

**Review by Indy2222 — CHANGES_REQUESTED** [review]
Could you change commit description (not the first line) so it includes `Fixes #142` to properly reference the issue (so it is automatically closed).

File: crates/spawner/src/draft.rs (hunk 2 above)
Comment (Indy2222) [reply crates/spawner/src/draft.rs]: Cool, I like the current solution!

**Review by Indy2222 — APPROVED** [review]

**Pull request merged**